src/core/elevation/qgsprofilesourceregistry.cpp

```diff
@@ -0,0 +1,45 @@
+/***************************************************************************
+  qgsprofilesourceregistry.cpp
+  --------------------------------------
+  Date                 : April 2024
+  Copyright            : (C) 2024 by Germán Carrillo
+  Email                : german at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsprofilesourceregistry.h"
+
+#include "qgsabstractprofilesource.h"
+
+QgsProfileSourceRegistry::QgsProfileSourceRegistry()
+{
+
+}
+
+QgsProfileSourceRegistry::~QgsProfileSourceRegistry()
+{
+  qDeleteAll( mSources );
+}
+
+QList< QgsAbstractProfileSource * > QgsProfileSourceRegistry::profileSources() const
+{
+  return mSources;
```

Comment:
I am not sure if the registry should return all the sources.
Shouldn't it return a clone of a source if you ask for a given id?

That's the way other registries are working

Reply:
I took [QgsExternalStorageRegistry](https://api.qgis.org/api/3.36/classQgsExternalStorageRegistry.html) as a basis, but I guess I should've had a look at other registries as well.

The rationale behind this is that we need to get the full list in order to iterate the sources in `QgsElevationProfileCanvas`, see https://github.com/gacarrillor/QGIS/pull/41/files#diff-7858eaf011cbfda84a6f705b8e0c0f64b95b113bf9a602d2fe6aa51548495862R848